src/main/java/org/junitpioneer/jupiter/StdIoExtension.java

```diff
@@ -61,11 +90,34 @@ private Object getOut() {
 	private String[] getSourceValuesFromAnnotation(ExtensionContext context) {
 		return context
 				.getTestMethod()
-				.map(method -> method.getAnnotation(StdInSource.class))
-				.map(StdInSource::value)
+				.map(method -> method.getAnnotation(StdIo.class))
+				.map(StdIo::value)
 				.orElseThrow(() -> new ParameterResolutionException(
-					format("Can not resolve test method parameter %s. Method has to be annotated with '%s'",
-						StdIn.class.getName(), StdInSource.class.getName())));
+					format("Can not resolve test method parameter %s. Method has to be annotated with '%s'.",
+						StdIn.class.getName(), StdIo.class.getName())));
+	}
+
+	@Override
+	public void beforeTestExecution(ExtensionContext context) {
+		Optional<Method> method = context.getTestMethod();
+		if (method.isPresent()) {
+			if (method.get().getAnnotation(StdIo.class) == null) {
+				throw new ExtensionConfigurationException(
+					format("StdIoExtension is active but no %s annotation was found.", StdIo.class.getName()));
+			}
+			List<Class<?>> params = Arrays.asList(method.get().getParameterTypes());
+			if (!params.contains(StdIn.class) && !params.contains(StdOut.class)) {
+				throw new ExtensionConfigurationException(
+					format("Method is annotated with %s but no %s or %s parameters were found.", StdIo.class.getName(),
+						StdIn.class.getName(), StdOut.class.getName()));
+			}
+			if (!String.join("", method.get().getAnnotation(StdIo.class).value()).isEmpty()
+					&& !params.contains(StdIn.class)) {
+				throw new ExtensionConfigurationException(format(
+					"Method has no %s parameter but input sources were provided in the %s annotation (Did you forget to add a test parameter?).",
+					StdIn.class.getName(), StdIo.class.getName()));
```

Comment:
I don't get this. The if tests if the value array is *not* empty and if the parameter does *not* contain `StdIn`, but the error messages says that there are parameter but `StdIn` was * not* provided. Which is totally fine, if `StdOut` was provided and the values are the expected written values.

Did you miss to add the check for `StdOut`, because you want to check if there is a) an annotation with values but b) neither `StdIn` or `StdOut` parameter?

Reply:
If the array is not empty, then you provided values for `StdIn` - which is incorrect if you don't have and `StdIn` parameter.